### PR TITLE
feat(fuse-network): integrated fuse in wallet

### PR DESCRIPTION
**Pull request by developerfred**
Fuse Network Integration
- [x] https://github.com/liquality/chainabstractionlayer/issues/446

### CODE REVIEW

File: src/views/CustomToken.vue [hunk 1]
```diff
@@ -62,6 +62,15 @@
                    ARBITRUM
                 </a>
               </li>
+              <li>
+                <a class="dropdown-item"
+                   id="arbitrum_chain"
```
[the remark this answers is not in the document]
Comment (developerfred) [reply src/views/CustomToken.vue:67]: thanks, fixed

File: src/store/factory/client.js [hunk 2]
```diff
@@ -207,6 +206,16 @@ function createArbitrumClient (asset, network, mnemonic, indexPath = 0) {
   return createEthereumClient(asset, network, arbitrumNetwork, rpcApi, scraperApi, feeProvider, mnemonic, 'default', indexPath)
 }
 
+function createFuuseClient (asset, network, mnemonic, indexPath = 0) {
+  const isTestnet = network === 'testnet'
+  const fuseNetwork = ChainNetworks.fuse[network]
+  const rpcApi = isTestnet ? 'https://rpc.fuse.io/' : ' https://fuse-mainnet.gateway.pokt.network/'
```
[the remark this answers is not in the document]
Comment (developerfred) [reply src/store/factory/client.js:212]: ok I wiil put testnet RPC

File: src/utils/fees.js [hunk 3]
```diff
@@ -12,7 +12,8 @@ const SEND_FEE_UNITS = {
   SOL: 1000000,
   MATIC: 21000,
   ERC20: 90000,
-  ARBETH: 620000
+  ARBETH: 620000,
+  FUSE: 1000000000
```
Comment (monokh) [inline src/utils/fees.js:16]: 1000000000 gas for send? Is that correct?

**Review by monokh — CHANGES_REQUESTED** [review]
could you provide some instructions how to test? For example where to get some testnet funds?

File: src/utils/fees.js (hunk 3 above)
Comment (developerfred) [reply src/utils/fees.js:16]: I fixed it: https://docs.fuse.io/the-fuse-chain/fuse-token#fees

**Comment by monokh** [comment]
@developerfred  Sorry I still can't figure out how to test this on testnet? Do you have a faucet?

File: src/build.config.js [hunk 4]
```diff
@@ -18,7 +18,8 @@ export default {
       'PWETH',
       'ARBETH',
       'FISH',
-      'SOL'
+      'SOL',
+      'FUSE'
     ],
     testnet: [
       'BTC',
```
Comment (monokh) [inline src/build.config.js:25]: Could you also add fuse to testnet so we can test it ?

File: src/build.config.js [hunk 5]
```diff
@@ -18,7 +18,8 @@ export default {
       'PWETH',
       'ARBETH',
       'FISH',
-      'SOL'
+      'SOL',
+      'FUSE'
```
Comment (monokh) [inline src/build.config.js:22]: Please remove fuse from the default assets,
only assets which we have vetted very closely are added to the user's wallet by default
Users can still use fuse by enabling it in Settings -> Manage assets

File: src/utils/fees.js [hunk 6]
```diff
@@ -12,7 +12,8 @@ const SEND_FEE_UNITS = {
   SOL: 1000000,
   MATIC: 21000,
   ERC20: 90000,
-  ARBETH: 620000
+  ARBETH: 620000,
+  FUSE: 0000000001
```
Comment (monokh) [inline src/utils/fees.js:16]: Does this mean just `1`? I'm surprised that the gas in fuse is not same as eth for sending (21000)

File: src/views/Receive.vue [hunk 7]
```diff
@@ -113,7 +114,8 @@ export default {
           NEAR: { name: 'NEAR', url: '' },
           SOL: { name: 'SOLANA', url: 'https://solfaucet.com/' },
           MATIC: { name: 'MATIC', url: 'https://faucet.matic.network/' },
-          ARBETH: { name: 'ARBETH', url: 'https://faucet.rinkeby.io/' }
+          ARBETH: { name: 'ARBETH', url: 'https://faucet.rinkeby.io/' },
+          FUSE: { name: 'FUSE', url: '' }
```
Comment (monokh) [inline src/views/Receive.vue:118]: If there is a faucet to get fuse, please add it here

**Review by monokh — CHANGES_REQUESTED** [review]

**Comment by monokh** [comment]
Also `ethereum-networks` needs to now be updated to 1.6.1. This version has the fuse networks

**Comment by monokh** [comment]
image
I don't see the fuse icon
and there is an error
```
Background.js:234 TypeError: Cannot read property 'name' of undefined
    at createEthereumClient (webpack:///./src/store/factory/client.js?:162:43)
    at createFuuseClient (webpack:///./src/store/factory/client.js?:318:10)
    at createClient (webpack:///./src/store/factory/client.js?:329:42)
```

I think this causes not to be able to use fuse

File: src/utils/fees.js (hunk 6 above)
Comment (developerfred) [reply src/utils/fees.js:16]: ref: https://explorer.fuse.io/tx/0xbb1cdab2f547b649eaca6ccc7b241bd643b89786602fa75b62306ca3007c197c/logs

Screenshot 2021-09-07 at 14 04 00

File: src/build.config.js (hunk 5 above)
Comment (developerfred) [reply src/build.config.js:22]: ok, done.

File: src/views/Receive.vue (hunk 7 above)
Comment (developerfred) [reply src/views/Receive.vue:118]: I will need to make a website and a faucet bot. I will try to implement a poc quickly.

File: src/build.config.js (hunk 4 above)
Comment (developerfred) [reply src/build.config.js:25]: > I think this causes not to be able to use fuse

how did you get to this error? I would like to simulate it.

File: src/views/Receive.vue (hunk 7 above)
Comment (developerfred) [reply src/views/Receive.vue:118]: I'm going to place the discord invite while I build the bot and website.

File: src/utils/fees.js (hunk 6 above)
Comment (monokh) [reply src/utils/fees.js:16]: 1 Gwei is the gas price, the gas limit in this screenshot is `91,366` This is what is meant by "fee units" in our case.
And the tx you have presented does not look like a regular TX, but a smart contract call. "SEND_FEE_UNITS" is for the most basic, sending Fuse from one address to another, i think it's probably 21000 like the other ethereum chains but you should try it and see
Gas price is always retrieved from the nodes using eth_gasPrice or another api

**Review by monokh — CHANGES_REQUESTED** [review]
You need to update the `@liquality/cryptoassets` and `@liquality/ethereum-networks` packages with the latest

Still don't see the icon for fuse

The integration still does not work and i get the error in the console from before. My fuse address does not display

I found this misspelling on the chainabstractionlayer work: 
`const fuse_mainnent: EthereumNetwork = {`

File: src/utils/fees.js (hunk 6 above)
Comment (developerfred) [reply src/utils/fees.js:16]: Thanks I will use ```eth_gasPrice```

**Comment by developerfred** [comment]
> You need to update the `@liquality/cryptoassets` and `@liquality/ethereum-networks` packages with the latest
> 
> Still don't see the icon for fuse
> 
> The integration still does not work and i get the error in the console from before. My fuse address does not display
> 
> I found this misspelling on the chainabstractionlayer work:
> `const fuse_mainnent: EthereumNetwork = {`

Gas Price = 1 Gwei check: https://docs.fuse.io/the-fuse-chain/network-overview-and-key-metrics

**Comment by developerfred** [comment]
> You need to update the `@liquality/cryptoassets` and `@liquality/ethereum-networks` packages with the latest
> 
> Still don't see the icon for fuse
> 
> The integration still does not work and i get the error in the console from before. My fuse address does not display
> 
> I found this misspelling on the chainabstractionlayer work:
> `const fuse_mainnent: EthereumNetwork = {`

done ✅

**Comment by monokh** [comment]
Still not working. Don't see fuse in "Manage Assets", and last time i tested the address was not displaying.

Can you test the wallet from a clean install and make sure you can do the following: 
- Enable the asset
- Get the FUSE address
- Send FUSE 
- Receive FUSE

**Comment by monokh** [comment]
Newer PR for this exists

**Pull request closed without merging**